OneCommunityGlobal/HighestGoodNetworkApp
# Luis edit warning description

# Description
Allow owners to edit a current warning description. 
Only owners are able to edit a warning description. 
Upon editing a warning the remaining warnings become uneditable until save is triggered.

## Related PRS (if any):
This frontend PR is related to the #[806](https://github.com/OneCommunityGlobal/HGNRest/pull/806) backend PR.


## Main changes explained:
- Inside of `WarningTrackerModal.jsx` I changed the p tag to input passing the styles and added the functionality needed.
- The functionality allows owners to edit by changing the same.
- Next a save button was added to trigger the actual save and put action sending the data to the backend.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log in as an owner account
4. Upon loading, click the "+/-" button to load all the current warnings. 
5. Upon loading, create a test, warning and submit it. 
6. Click the newly created warning, and edit it. 
7. Once edited, click the 'save' button to send the updated warning. 
8. View the user's warnings and should display the newly edited warning. Also, the `WarningTrackerModal.jsx`will have the updated list as well. 

## Screenshots or videos of changes:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/fbbfc65f-10ca-492f-8a3c-52b4393cd3e5


## Note:
Include the information the reviewers need to know.
There is a bug, which when creating a new warning, it doesn't allow you to add spaces, I'll fix the issue in this PR #2037
